services/frontend/src/components/basicInfo/BasicInfoView.jsx

```diff
@@ -328,21 +270,80 @@ const BasicInfoView = ({
     return buttons;
   };
 
+  /*
+   * Get Action Ribbon Button
+   *
+   * Retrieves the ribbon button for adding/removing a user from one's circle
+   */
+  const getActionRibbonBtn = () => {
+    if (urlID === userID) {
+      return (
+        <Button
+          aria-label={intl.formatMessage({ id: "edit.profile" })}
+          block
+          className="ribbon-btn"
+          icon={<EditOutlined aria-hidden="true" className="ribbon-btn-icon" />}
+          onClick={() => {history.push("/profile/edit/primary-info")}}
+          tabIndex={0}
+          type="link"
+        >
+          <FormattedMessage id="edit.profile" />
+        </Button>
+      );
+    }
+
+    return (
+      <Button
+        aria-label={
+          connectionStatus ? (
+            intl.formatMessage({ id: "remove.connection" })
+          ) : (
+            intl.formatMessage({ id: "add.connection" })
+          )
+        }
+        block
+        className="ribbon-btn"
+        icon={
+          connectionStatus ? (
+            <UserDeleteOutlined aria-hidden="true" className="ribbon-btn-icon" />
+          ) : (
+            <UserAddOutlined aria-hidden="true" className="ribbon-btn-icon" />
+          )
+        }
+        onClick={changeConnection}
+        tabIndex={0}
+        type="link"
+      >
+        {connectionStatus ? (
+          <FormattedMessage id="remove.connection" />
+        ) : (
+          <FormattedMessage id="add.connection" />
+        )}
+      </Button>
+    );
+  }
+
   return (
-    <Card actions={generateActions()} id="card-profile-basic-info">
-      {generateProfileHeader()}
-      <Row>
-        <Col lg={12} xs={24}>
-          {generateInfoList(getContactInfo())}
-        </Col>
-        <Col lg={12} xs={24}>
-          {generateInfoList(getWorkInfo())}
-        </Col>
-      </Row>
-      <Row className="rowTopSplitter">
-        <Col span={24}>{generateInfoList(generateTeamInfo())}</Col>
-      </Row>
-    </Card>
+    <Badge.Ribbon
```

Comment:
im thinking we should create a component for the "friendship ribbon" since it has a series of repeated code. what do you think?

Reply:
That's a good idea. I created the component and moved over the repetitive code. Let me know what you think.